releasenotes/notes/add-integration-guardrail-versions-when-patching-bd5897789bb046c6.yaml

```diff
@@ -0,0 +1,5 @@
+---
+other:
+  - |
+    Adds guardrail versions to integrations which prevent patching of incompatible versions via a new `_supported_versions` function.
```

Comment:
make sure to write the release note for our customers not our devs. so you should describe the new change, the new env variable, that it is disabled by default for now, etc.

can also leave a note explicitly about the support range added for aiobotocore and fastapi, like "when enabled prevent trying to patch fastapi <2.3.5", etc.